crates/bevy_ui/src/focus.rs

```diff
@@ -1,350 +1,5 @@
-};
-use bevy_ecs::{
-    change_detection::DetectChangesMut,
-    entity::Entity,
-    prelude::{Component, With},
-    query::QueryData,
-    reflect::ReflectComponent,
-    system::{Local, Query, Res},
-};
-use bevy_input::{mouse::MouseButton, touch::Touches, ButtonInput};
-use bevy_math::{Rect, Vec2};
-use bevy_reflect::{std_traits::ReflectDefault, Reflect};
-use bevy_render::{camera::NormalizedRenderTarget, prelude::Camera, view::ViewVisibility};
-use bevy_transform::components::GlobalTransform;
-use bevy_utils::HashMap;
-use bevy_window::{PrimaryWindow, Window};
-
-use smallvec::SmallVec;
-
-#[cfg(feature = "serialize")]
-use bevy_reflect::{ReflectDeserialize, ReflectSerialize};
-
-/// Describes what type of input interaction has occurred for a UI node.
-///
-/// This is commonly queried with a `Changed<Interaction>` filter.
-///
-/// Updated in [`ui_focus_system`].
-///
-/// If a UI node has both [`Interaction`] and [`ViewVisibility`] components,
-/// [`Interaction`] will always be [`Interaction::None`]
-/// when [`ViewVisibility::get()`] is false.
-/// This ensures that hidden UI nodes are not interactable,
-/// and do not end up stuck in an active state if hidden at the wrong time.
-///
-/// Note that you can also control the visibility of a node using the [`Display`](crate::ui_node::Display) property,
-/// which fully collapses it during layout calculations.
-///
-/// # See also
-///
-/// - [`ButtonBundle`](crate::node_bundles::ButtonBundle) which includes this component
-/// - [`RelativeCursorPosition`] to obtain the position of the cursor relative to current node
-#[derive(Component, Copy, Clone, Eq, PartialEq, Debug, Reflect)]
-#[reflect(Component, Default, PartialEq, Debug)]
-#[cfg_attr(
-    feature = "serialize",
-    derive(serde::Serialize, serde::Deserialize),
-    reflect(Serialize, Deserialize)
-)]
-pub enum Interaction {
-    /// The node has been pressed.
-    ///
-    /// Note: This does not capture click/press-release action.
-    Pressed,
-    /// The node has been hovered over
-    Hovered,
-    /// Nothing has happened
-    None,
-}
-
-impl Interaction {
-    const DEFAULT: Self = Self::None;
-}
-
-impl Default for Interaction {
-    fn default() -> Self {
-        Self::DEFAULT
-    }
-}
-
-/// A component storing the position of the mouse relative to the node, (0., 0.) being the top-left corner and (1., 1.) being the bottom-right
-/// If the mouse is not over the node, the value will go beyond the range of (0., 0.) to (1., 1.)
-///
-/// It can be used alongside [`Interaction`] to get the position of the press.
-///
-/// The component is updated when it is in the same entity with [`Node`].
-#[derive(Component, Copy, Clone, Default, PartialEq, Debug, Reflect)]
-#[reflect(Component, Default, PartialEq, Debug)]
-#[cfg_attr(
-    feature = "serialize",
-    derive(serde::Serialize, serde::Deserialize),
-    reflect(Serialize, Deserialize)
-)]
-pub struct RelativeCursorPosition {
```

Comment:
We could keep the `RelativeCursorPosition` component for now and just update it in `update_hover_status` or something. It would make the migration simpler.